### PR TITLE
Add hibernate.query.in_clause_parameter_padding as a supported configuration parameter

### LINKED ISSUE #29196: Add "hibernate.query.in_clause_parameter_padding" as a supported configuration parameter
### Description

We're using `quarkus.hibernate-orm.unsupported-properties."hibernate.query.in_clause_parameter_padding"=true` to improve the statement caching for heavily used IN clause queries.

I propose to add `hibernate.query.in_clause_parameter_padding` as a supported Quarkus parameter.

Thanks!

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by zedbeit**
…uration parameter.   Fix #29196

### CODE REVIEW

**Comment by zedbeit** [comment]
My attempt to fix #29196, @gsmet pls review.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java [hunk 1]
```diff
@@ -329,9 +329,18 @@ public enum NullOrdering {
         @ConfigItem(defaultValue = "none")
         public NullOrdering defaultNullOrdering;
 
+        /**
+         * Improve statement caching for heavily used IN clause queries
```
Comment (gsmet) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:333]:
```suggestion
         * Enables IN clause parameter padding which improves statement caching.
```

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java [hunk 2]
```diff
@@ -329,9 +329,18 @@ public enum NullOrdering {
...
+         *
+         * @asciidoclet
+         */
+        @ConfigItem(defaultValue = "true")
```
Comment (gsmet) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:337]: From what I can see, you are changing the default behavior here. I'm not sure if we want to do that.

@Sanne any opinion on this?

**Review by gsmet — CHANGES_REQUESTED** [review]
Overall, it looks good, thanks. I'm just wondering about the change in default value.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java (hunk 2 above)
Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:337]: I'm not sure - it's not particularly dangerous, might be fine and I agree it's often a good idea to enable it.

Wondering if we should switch the default in ORM6 (it's still false currently), I'll as the team.

I'm inclined to enable it.

Comment (gastaldi) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:337]: How about declaring this attribute as `Optional<Boolean>` and set it only if it's explicitly set?

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java [hunk 3]
```diff
@@ -329,9 +329,18 @@ public enum NullOrdering {
         @ConfigItem(defaultValue = "none")
         public NullOrdering defaultNullOrdering;
 
+        /**
+         * Improve statement caching for heavily used IN clause queries
+         *
+         * @asciidoclet
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:335]: This is not needed

**Review by gastaldi — COMMENTED** [review]
It would be nice to have a test too

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java (hunk 2 above)
Comment (gsmet) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:337]: Given we are leaning towards making the default true, that wouldn't make much sense.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java [hunk 4]
```diff
@@ -1072,6 +1072,10 @@ private static void producePersistenceUnitDescriptorFromConfig(
         descriptor.getProperties().setProperty(AvailableSettings.DEFAULT_NULL_ORDERING,
                 persistenceUnitConfig.query.defaultNullOrdering.name().toLowerCase(Locale.ROOT));
 
+        if (persistenceUnitConfig.query.inClauseParameterPadding) {
+            descriptor.getProperties().setProperty(AvailableSettings.IN_CLAUSE_PARAMETER_PADDING, "true");
+        }
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:1077]: Shouldn't this be set if it's `false` too?

**Comment by zedbeit** [comment]
Thanks everyone for the review, I've made the suggested changes.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java [hunk 5]
```diff
@@ -1072,6 +1072,12 @@ private static void producePersistenceUnitDescriptorFromConfig(
         descriptor.getProperties().setProperty(AvailableSettings.DEFAULT_NULL_ORDERING,
                 persistenceUnitConfig.query.defaultNullOrdering.name().toLowerCase(Locale.ROOT));
 
+        if (persistenceUnitConfig.query.inClauseParameterPadding) {
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:1075]: This `if` is not necessary, as you can use the attribute value directly

Comment (zedbeit) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:1075]: Not sure I get what you mean

File: extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java [hunk 6]
```diff
@@ -306,6 +306,12 @@ private static ParsedPersistenceXmlDescriptor generateReactivePersistenceUnit(
         desc.getProperties().setProperty(AvailableSettings.DEFAULT_NULL_ORDERING,
                 persistenceUnitConfig.query.defaultNullOrdering.name().toLowerCase());
 
+        if (persistenceUnitConfig.query.inClauseParameterPadding) {
```
Comment (gastaldi) [inline extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java:309]:
```suggestion
         desc.getProperties().setProperty(AvailableSettings.IN_CLAUSE_PARAMETER_PADDING, String.valueOf(persistenceUnitConfig.query.inClauseParameterPadding));
```

Comment (zedbeit) [reply extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java:309]: Cool, this is much cleaner, thanks.

**Comment by gastaldi** [comment]
Can you squash the commits and force push to your branch?

**Comment by zedbeit** [comment]
alright

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java [hunk 7]
```diff
@@ -329,9 +329,18 @@ public enum NullOrdering {
         @ConfigItem(defaultValue = "none")
         public NullOrdering defaultNullOrdering;
 
+        /**
+         * Enables IN clause parameter padding which improves statement caching.
+         *
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:334]: Can you also remove the extra line breaks?

Comment (zedbeit) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:334]: Just seeing this after I pushed, should I still make this change?

Comment (gastaldi) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:334]: Yes

Comment (zedbeit) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java]: done.

**Review by gastaldi — APPROVED** [review]
LGTM, good job!

**Review by gsmet — APPROVED** [review]
The test failure doesn't look related.

**Comment by gsmet** [comment]
Merged, thanks!

I will include it in 2.15 given it's a relatively simple change and if there is an issue, the workaround is easy.

**Pull request merged**